src/pytypes/onefuzztypes/models.py

```diff
@@ -799,6 +799,11 @@ class NetworkConfig(BaseModel):
     subnet: str = Field(default="10.0.0.0/16")
 
 
+class NetworkSecurityGroupConfig(BaseModel):
+    allow_service_tags: Optional[List[str]]
+    allow_IPs: List[str] = Field(default=[])
```

Comment:
Python attribute names are conventionally always lowercase. See [here](https://www.python.org/dev/peps/pep-0008/#method-names-and-instance-variables).

I wonder if it would make more sense to write `allowed_` instead of `allow_`. The latter sounds more like an imperative, and at a glance, I expect the value to be boolean. To me, using the adjective `allowed_` builds more of a noun phrase, which better matches the fact that this field is a "list of allowed IPs". WDYT?

If we applied both of my suggestions, this would become:
```suggestion
    allowed_ips: List[str] = Field(default=[])
```

Reply:
Oh, and I just noted that in other models, we have `allowed_aad_tenants` for a list-valued field, so my suggestion is more consistent with that.

Reply:
Okay updating!

Reply:
Also, warning, there is a general Python gotcha here we should be careful of: the default value `[]` is mutable. This can cause issues in some situations when you mutate the default value.

Example with plain old functions:
```python
>>> def foo(values=[], x=None):
...     if x is not None:
...             values.append(x)
...     print(values)
...

>>> foo()
[]

# This will mutate the specific instance of `list` that is `foo`'s default arg!
>>> foo(x='hi')
['hi']

# We're printing the default value, which was mutated.
# Our default value isn't a new, empty list. It is a reference to a specific list.
>>> foo()
['hi']
```
Pydantic appears to handle this for us by deep-copying under the hood, but the fact that this works is NOT a given.

Reply:
(here's where Pydantic has code to do what we really want in this situation: https://github.com/samuelcolvin/pydantic/blob/master/pydantic/fields.py#L375-L376)

Reply:
Note, `Field(default_factory=list)` will work here.

Reply:
We use this pattern in the CLI a few times:

````
onefuzz/backend.py
93:    features: Set[str] = Field(default_factory=set)

onefuzz/job_templates/cache.py
26:    entries: Dict[str, EndpointCache] = Field(default_factory=dict)
```

Reply:
Updated! Thanks for the explanation!